src/Auth/DatabaseAuthenticator.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Auth;
+
+use Tempest\Auth\Exceptions\MissingIdentifiableException;
+use Tempest\Database\Query;
+use Tempest\Http\Session\Session;
+use function Tempest\make;
+
+final class DatabaseAuthenticator implements Authenticator
+{
+    use AuthenticatorWithSession;
+
+    public function __construct(
+        protected AuthConfig $config,
+        Session $session,
+    ) {
+        $this->session = $session;
+    }
+
+    public function login(Identifiable $identifiable): void
+    {
+        $this->createSession($identifiable);
+    }
+
+    public function logout(): void
+    {
+        $this->destroySession();
+    }
+
+    /**
+     * @throws MissingIdentifiableException
+     */
+    public function user(): Identifiable|null
```

Comment:
IMO this method doesn't belong here. We should split the concern of resolving an identifiable and logging them in. So, in my mind, we'd have `DatabaseIdentifier` and `SessionAuthenticator`.

We could add more identifiers in the future like, for example, `GithubOAuthIdentifier`, and more authenticators (I actually can't come up with a proper example besides session atm)